src/shared/components/data-table/components/main/components/table-row/TableRow.test.tsx

```diff
@@ -0,0 +1,137 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import {
+  TableContext,
+  type TableContextType
+} from 'src/shared/components/data-table/DataTable';
+
+import TableRow from './TableRow';
+import { defaultDataTableState } from 'src/shared/components/data-table/dataTableReducer';
+import { createDataTableSampleData } from 'src/shared/components/data-table/DataTable.test';
+
+const { columns, data } = createDataTableSampleData(1, 10);
+
+const defaultProps = {
+  ...defaultDataTableState,
+  data,
+  dispatch: jest.fn(),
+  columns,
+  rows: data.map((row, index) => ({
+    rowId: index,
+    cells: row
+  }))
+};
+
+describe('<TableRow />', () => {
+  const renderTableRow = (props: Partial<TableContextType> = {}) =>
+    render(
+      <TableContext.Provider value={{ ...defaultProps, ...props }}>
+        <table>
+          <tbody>
+            <TableRow
+              rowId={String(defaultProps.rows[0].rowId)}
+              rowData={defaultProps.rows[0].cells}
+            />
+          </tbody>
+        </table>
+      </TableContext.Provider>
+    );
+
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders without error', () => {
+    container = renderTableRow().container;
+    expect(() => container).not.toThrow();
+  });
+
+  it('respects hiddenRowIds and renders nothing', () => {
+    container = renderTableRow({
+      hiddenRowIds: { [defaultProps.rows[0].rowId]: true }
+    }).container;
+
+    const tableRowElement = container.querySelector('tr');
+    expect(tableRowElement).toBeFalsy();
+  });
+
+  it('respects hiddenColumnIds', () => {
+    container = renderTableRow().container;
+
+    const [firstCellContent, secondCellContent] = defaultProps.rows[0].cells;
+    let firstCellElement = container.querySelector('td');
+    expect(firstCellElement?.textContent).toBe(firstCellContent);
+
+    container = renderTableRow({
+      hiddenColumnIds: { [defaultProps.columns[0].columnId]: true }
+    }).container;
+
+    firstCellElement = container.querySelector('td');
+    expect(firstCellElement?.textContent).toBe(secondCellContent);
+  });
+
+  it('respects individual column renderer prop', () => {
+    const columns = [...defaultProps.columns];
+    const renderer = jest.fn();
+
+    const indexOfColumnWithRenderer = 0;
+    columns[indexOfColumnWithRenderer].renderer = renderer;
+
+    container = renderTableRow({
+      columns
+    }).container;
+
+    expect(renderer).toBeCalledWith({
+      rowData: defaultProps.rows[0].cells,
+      rowId: String(defaultProps.rows[0].rowId),
+      cellData: defaultProps.rows[0].cells[indexOfColumnWithRenderer]
```

Comment:
I pass the rowData because the data displayed in a particular column could be derived by combining multiple columns.
I think it can be updated to be the entire table data when necessary.